### PR TITLE
Updated html_root_url for compiler crates

### LINKED ISSUE #77103: Oudated html_root_url for compiler crates
https://doc.rust-lang.org/nightly/ should be https://doc.rust-lang.org/nightly/nightly-rustc/

### PR DESCRIPTION
**Pull request by duckymirror**
Closes #77103

r? @jyn514

### CODE REVIEW

File: compiler/rustc_apfloat/src/lib.rs [hunk 1]
```diff
@@ -30,7 +30,7 @@
 //!
 //! This API is completely unstable and subject to change.
 
-#![doc(html_root_url = "https://doc.rust-lang.org/nightly/")]
+#![doc(html_root_url = "https://doc.rust-lang.org/nightly-rustc/")]
```
Comment (Urgau) [inline compiler/rustc_apfloat/src/lib.rs:33]: All of them should be:
```suggestion
#![doc(html_root_url = "https://doc.rust-lang.org/nightly/nightly-rustc/")]
```

Comment (duckymirror) [reply compiler/rustc_apfloat/src/lib.rs:33]: I feel _very_ stupid now. I'll try that again.

**Comment by jyn514** [comment]
r=me with CI passing

**Pull request merged**